### ISSUE TITLE
axe-core ignores elements that have position: fixed or position:absolute

### ISSUE THREAD

**Issue by marianntapfer** [dequelabs/axe-core#3996]
### Product

axe-core

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
All elements in should get tested with tests relevant to them. If a button inside a modal had color contrast issues it should be reported.

#### Actual
It seems like elements that have position: fixed  or position: absolute are not tested. 

#### How to Reproduce
I made a simple [reproduction](https://github.com/marianntapfer/brick-ui/pull/5) in this test component library.

#### Additional context
 I got then issue when using Storybook's addon-a11y and axe-playwright in Storybook testrunner.

**Comment by straker** [comment]
Thanks for the issue. I'm able to reproduce it and have determined that it's a bug in the `axe.commons.dom.isVisibleOnScreen` function (specifically the overflow hidden visibility method). The `body` has `overflow: hidden` but that doesn't make any content outside the bounding box of the body element visually hidden (like it would for any other node). So when we test if the element is inside the bounding box of the overflow hidden container it returns `false`, so is not treated as visible.

Here's a minimum setup to trigger the problem.

```html
<body style="overflow: hidden">
  <button>Top</button>
  
  <div style="position: absolute; top: 100px;">
    <button style="color: #aaa">Error</button>
  </div>
</body>
```
